src/EFCore.MySql/Query/ExpressionVisitors/Internal/MySqlHavingExpressionVisitor.cs

```diff
@@ -50,9 +50,18 @@ protected virtual Expression VisitSelect(SelectExpression selectExpression)
                     // Hack around the fact, that EF Core does not allow us to add our own ProjectionExpression with an alias.
                     // We can however indirectly control the alias, by first setting a ColumnExpression, whose name is then used as the base
                     // for the alias, and then update it afterwards with the actual expression we are interested in, when it gets visited.
+                    //
+                    // Unfortunately, EF Core 6 changed its internal algorithm a bit and now visits the expression before it considers its
+                    // name (if a ColumnExpression), so we now have to wrap our HatSwappingColumnExpression inside of another (dummy)
+                    // HatSwappingColumnExpression, so that the outer one can get visited/peeled off by EF Core immediately without
+                    // touching the inner one, which can then present the actual expression when visited.
                     selectExpression.AddToProjection(
                         new HatSwappingColumnExpression(
-                            havingExpression,
+                            new HatSwappingColumnExpression(
+                                havingExpression,
+                                alias,
+                                havingExpression.Type,
+                                havingExpression.TypeMapping),
                             alias,
                             havingExpression.Type,
                             havingExpression.TypeMapping));
```

Comment:
Why not just
```cs
                    var projectionIndex = subQuery.AddToProjection(havingExpression);
                    var alias = subQuery.Projection[projectionIndex].Alias;
```

We would loose any control over the name, but no longer risk having to adopt this code every other version of EF Core

Reply:
If it works and is simpler or has less maintenance overhead, go ahead and open a PR!

Reply:
Created #1681